packages/nextjs/app/page.tsx

```diff
@@ -19,6 +20,77 @@ const Home: NextPage = () => {
     }
   }
 
+  const { address } = useAccount();
+
+  const { data: Ens } = useEnsName({
+    address,
+    chainId: 1,
+  });
+
+  const { data: allowStatus } = useScaffoldContractRead({
+    contractName: "BatchRegistry",
+    functionName: "allowList",
+    args: [address],
+  });
+
+  const { data: checkedInStatus } = useScaffoldContractRead({
+    contractName: "BatchRegistry",
+    functionName: "yourContractAddress",
+    args: [address],
+  });
+  const CheckStatus = useCallback(() => {
+    if (!address) return <span>Please connect your wallet to see feature more details.</span>;
+
+    const taskUrl = "https://github.com/BuidlGuidl/batch3.buidlguidl.com/issues";
+    const issueUrl = taskUrl + "/9";
+
+    if (!allowStatus)
+      return (
+        <span>
+          Heard about our challenge take a look at{" "}
+          <Link
+            href="https://speedrunethereum.com"
+            target="_blank"
+            className="text-indigo-500 hover:text-indigo-600 transition-colors duration-300"
+          >
+            speedrunethereum.
+          </Link>
+        </span>
+      );
+
+    if (allowStatus && !checkedInStatus)
```

Comment:
The `!checkedInStatus` will always be true because if you are not checkedIn  the `yourContractAddress` function from contract returns you zero address as string : "0x0000000000000000000000000000000000000000" and Boolean("0x00..") is true

Instead you should add strict check comparing both strings, something like : 

```typescript
const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000"
const isCheckedIn = checkedInStatus !== ZERO_ADDRESS
```

Nitpick: We could also rename `checkedInStatus` to better reperesent it an address which is being returned